code/modules/mob/mob_helpers.dm

```diff
@@ -22,6 +22,30 @@
 			return 0
 	return 1
 
+/mob/proc/get_screen_colour()
+
+/mob/proc/update_client_colour(var/time = 10) //Update the mob's client.color with an animation the specified time in length.
+	if(!client) //No client_colour without client. If the player logs back in they'll be back through here anyway.
+		return
+	client.colour_transition(get_screen_colour(), time = time) //Get the colour matrix we're going to transition to depending on relevance (magic glasses first, eyes second).
+
+/mob/living/carbon/human/get_screen_colour() //Fetch the colour matrix from wherever (e.g. eyes) so it can be compared to client.color.
+	. = ..()
+	if(.)
+		return .
+	var/obj/item/organ/internal/eyes/eyes = get_int_organ(/obj/item/organ/internal/eyes)
+	if(eyes && !eyes.robotic) //Activate/deactivate colourblindness on the mob. Works hand-in-hand with eye insertion/removal procs in code\modules\surgery\organs\organ_internal.dm.
+		if(disabilities & COLOURBLIND)
+			eyes.apply_colourblindness()
+		else if(eyes.colourmatrix && eyes.colourblind_darkview != initial(eyes.dark_view)) //Avoids giving colour vision to mobs with incurable non-genetic colourblindness (e.g. Farwas/Wolpins).
+			eyes.remove_colourblindness()
+
+	var/obj/item/clothing/glasses/worn_glasses = glasses
+	if(istype(worn_glasses) && worn_glasses.color_view) //Check to see if they got those magic glasses and they're augmenting the colour of what the wearer sees. If they're not, color_view should be null.
+		return worn_glasses.color_view
+	else if(eyes && eyes.colourmatrix) //If they're not, check to see if their eyes got one of them there colour matrices.
+		return eyes.colourmatrix

```

Comment:
And here, instead of `eyes.colourmatrix`, you'd use `eyes.get_colourmatrix()`. The code higher up in this proc that updates eyes on colorblindness changes wouldn't need to exist, because the helper procs would always return the appropriate values. This proc wouldn't even need to be aware that colorblindness exists.